### PR TITLE
Idea Project + Register Allocation + Reaching Definitions

### PR DESCRIPTION
**Pull request by lacinoire**
Cleaned up intellij integration (ignore generated files)
Denis' Implementation of Liveness & Register Allocation
Caros Implementation of Reaching Definitions

### CODE REVIEW

File: analysis/analysis/RegisterAllocationAnalysis.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    TrueLivenessAnalysis la;
+
+    public RegisterAllocationAnalysis(CompilationUnit cu, TrueLivenessAnalysis la) {
+        super(true);
+        this.cu = cu;
+        this.la = la;
+    }
+
+    public Map<Variable, Integer> visit(ProcedureCall procedureCall, Map<Variable, Integer> objectObjectMap) {
+        return null;
```
Comment (Flakebi) [inline analysis/analysis/RegisterAllocationAnalysis.java]: Did you test this when the C code contains a function call?
My guess would be that the analysis stops on the first call (though I’m not completely sure).

**Review by Flakebi — COMMENTED** [review]
Apart from my comment, it looks good.

File: analysis/analysis/RegisterAllocationAnalysis.java (hunk 1 above)
Comment (denis631) [reply analysis/analysis/RegisterAllocationAnalysis.java]: Did we handle procedure calls, when it comes to register analysis. I don't think we did, this is why I didn't do it, but I can just propagate the prev value, instead of passing null

Comment (Flakebi) [reply analysis/analysis/RegisterAllocationAnalysis.java]: Good point, I guess implementing calling conventions would be done in another compilation step so passing sounds good.

File: analysis/analysis/ReachingDefinitionsAnalysis.java [hunk 2]
```diff
@@ -41,14 +41,22 @@ public ReachingDefinitionsAnalysis(CompilationUnit cu) {
...
+                    }
+                    return res;
+                }
             );
+            System.out.println("remo " + variable.toString() + ", " + newDefinitions.size());
             newDefinitions
                 .add(new Tupel<Variable, Long>(variable, assignment.getDest().getId()));
         }
+        System.out.println(newDefinitions.size());
         return newDefinitions;
```
Comment (Flakebi) [inline analysis/analysis/ReachingDefinitionsAnalysis.java:60]: If the variable equals comparison works, I guess you can revert the changes to this file :)
